bravado/http_future.py

```diff
@@ -8,12 +8,33 @@
 from bravado.exception import HTTPError
 
 
+class FutureAdapter(object):
+    """
+    Mimics a :class:`concurrent.futures.Future` regardless of which client is
+    performing the request, whether it is synchronous or actually asynchornous.
+
+    This adapter must be implemented by all bravado clients such as FidoClient
+    or RequestsClient to wrap the object returned by their 'request' method.
```

Comment:
that's just the interface,
RequestsClient is using it [here](https://github.com/Yelp/bravado/pull/218/files#diff-0c505f6596e107bab9531f24ef030343R198)
```python
class RequestsFutureAdapter(FutureAdapter):
```

 and FidoClient is using it [here](https://github.com/Yelp/bravado/pull/218/files#diff-4e6e269b35d40d26025c53fbc082cc28R134)
```python
class FidoFutureAdapter(FutureAdapter):
```
:)